web/frontend/src/stores/persisted.ts

```diff
@@ -0,0 +1,78 @@
+import { writable, type Writable } from 'svelte/store';
+
+declare type Updater<T> = (value: T) => T;
+declare type StoreDict<T> = { [key: string]: Writable<T> }
+
+const stores: StoreDict<unknown> = {};
+
+type StorageType = 'local' | 'session'
+
+const getStorage = (type: StorageType) => {
+  return type === 'local' ? localStorage : sessionStorage;
+};
+
+/**
+ * Creates a writable store that persists in localStorage
+ * @param key The storage key
+ * @param initialValue The initial value to put in storage if no value exists.
+ * @param storageType 'local' or 'session'
+ */
+export const persisted = <T>(
+  key: string,
+  initialValue: T,
+  storageType: 'local' | 'session' = 'local'
+): Writable<T> => {
+  const browser = typeof window !== 'undefined' && typeof document !== 'undefined';
```

Comment:
Should we leave a TODO to use the SvelteKit browser flag?

Reply:
Good idea, will add